src/Persistence_matrix/include/gudhi/persistence_interval.h

```diff
@@ -0,0 +1,208 @@
+   * death value to @ref inf.
+   * 
+   * @param dim Dimension of the cycle.
+   * @param birth Birth value of the cycle.
+   */
+  Persistence_interval(dimension_type dim, event_value_type birth) : dim(dim), birth(birth), death(inf) {}
+
+  /**
+   * @brief Constructor. Initializes the stored dimension, the stored birth value and the stored
+   * death value to the given values.
+   * 
+   * @param dim Dimension of the cycle.
+   * @param birth Birth value of the cycle.
+   * @param death Death value of the cycle.
+   */
+  Persistence_interval(dimension_type dim, event_value_type birth, event_value_type death)
+      : dim(dim), birth(birth), death(death) {}
+
+  dimension_type dim;     /**< Dimension of the cycle.*/
+  event_value_type birth; /**< Birth value of the cycle. */
+  event_value_type death; /**< Death value of the cycle. */
+
+  inline friend std::ostream &operator<<(std::ostream &stream, const Persistence_interval &interval) {
+    stream << "[" << interval.dim << "] ";
+    stream << interval.birth << ", " << interval.death;
+    return stream;
+  }
+};
+
+}  // namespace persistence_matrix
+}  // namespace Gudhi
+
+namespace std {
+
+/**
+ * @ingroup persistence_matrix
+ *
+ * @brief Overload of `std::tuple_size` for @ref Gudhi::persistence_matrix::Persistence_interval.
+ * 
+ * @tparam dimension_type First template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
+ * @tparam event_value_type Second template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
+ */
+template <typename dimension_type, typename event_value_type>
+struct tuple_size<Gudhi::persistence_matrix::Persistence_interval<dimension_type, event_value_type> >
+    : std::integral_constant<std::size_t, 3> {};
+
+/**
+ * @ingroup persistence_matrix
+ *
+ * @brief Overload of `std::tuple_element` for @ref Gudhi::persistence_matrix::Persistence_interval.
+ * 
+ * @tparam I Index of the type to store: 0 for the birth value type, 1 for the death value type and 2 for the
+ * dimension value type.
+ * @tparam dimension_type First template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
+ * @tparam event_value_type Second template parameter of @ref Gudhi::persistence_matrix::Persistence_interval.
+ */
+template <std::size_t I, typename dimension_type, typename event_value_type>
+struct tuple_element<I, Gudhi::persistence_matrix::Persistence_interval<dimension_type, event_value_type> > {
+  static_assert(I < 3, "Value mismatch at argument 1 in template parameter list. Maximal possible value is 2.");
+
+  using type = typename std::conditional<I < 2, event_value_type, dimension_type>::type;
+};
+
+/**
+ * @ingroup persistence_matrix
+ *
+ * @brief Overload of `std::get` for @ref Gudhi::persistence_matrix::Persistence_interval.
+ * 
+ * @tparam I Index of the value to return: 0 for the birth value, 1 for the death value and 2 for the dimension.
```

Comment:
It seems confusing that the constructor uses the order dim-birth-death while the access is in order birth-death-dim. Can't we pick one consistent order? Or are you trying to be consistent with 2 existing, incompatible types, and this is some compromise?

Reply:
I initially did all the matrix, zigzag etc. with the order dim-birth-death in mind. But when I was looking at making it consistent with the output of cohomology when adding the `std::get` overloads, I saw that the tuples are in order birth-death-coeff. So I wanted the `std::get` to stick closer to that. I thought that it won't be much of a problem as the users using the constructors them-selves would probably not use the `std::get` to access the members and therefore not be confused. But I agree, it was mostly laziness on my side, as I didn't wanted to correct all the calls to the bars in the matrix to change the order... But I could do that if you find it really to confusing.

Reply:
I think I like the new consistent order better (I don't mind that printing is in a different order because the `[]` around the dimension don't make it look like a tuple).
A drawback is that we are probably less likely to be able to use the short versions of the constructor with 1 or 2 elements :shrug:

Reply:
Yes, I used the short version in zigzag and will have to use the long version now. But it is not the end of the world. Or I could go back to have three different constructors.